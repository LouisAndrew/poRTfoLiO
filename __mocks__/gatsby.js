const React = require('react')
const gatsby = jest.requireActual('gatsby')

const mockSocials = [
    { link: 'http://localhost', social_key: 'github' },
    { link: 'http://google.com', social_key: 'dribbble' },
]

const mockButtonText = 'Click me'
const mockContactContent = 'Content'
const mockContactHeadline = 'Headline'
const mockEmail = 'john@doe.com'

module.exports = {
    ...gatsby,
    graphql: jest.fn(),
    Link: jest.fn().mockImplementation(
        // these props are invalid for an `a` tag
        ({
            activeClassName,
            activeStyle,
            getProps,
            innerRef,
            ref,
            replace,
            to,
            ...rest
        }) =>
            React.createElement('a', {
                ...rest,
                href: to,
            })
    ),
    StaticQuery: jest.fn(),
    // can I mock return value here?
    useStaticQuery: jest.fn(() => ({
        socialData: {
            frontmatter: {
                social: mockSocials,
            },
        },
        emailData: {
            frontmatter: {
                contact: {
                    button_text: mockButtonText,
                    contact_content: mockContactContent,
                    contact_headline: mockContactHeadline,
                    email: mockEmail,
                },
            },
        },

        // forced to write the hero-page-query mock here.
        heroData: {
            frontmatter: {
                hero: {
                    hero_content: 'Hello, World!', // hard coded from hero.test.tsx
                    hero_headline: 'I am headline', // Also hard coded from hero.test.tsx
                },
            },
        },
        imgS: {
            frontmatter: {
                hero: {
                    hero_imgs: [
                        {
                            childImageSharp: {
                                fixed: {},
                            },
                        },
                    ],
                },
            },
        },
        imgM: {
            frontmatter: {
                hero: {
                    hero_imgs: [
                        {
                            childImageSharp: {
                                fixed: {},
                            },
                        },
                    ],
                },
            },
        },
        imgL: {
            frontmatter: {
                hero: {
                    hero_imgs: [
                        {
                            childImageSharp: {
                                fixed: {},
                            },
                        },
                    ],
                },
            },
        },
        imgXL: {
            frontmatter: {
                hero: {
                    hero_imgs: [
                        {
                            childImageSharp: {
                                fixed: {},
                            },
                        },
                    ],
                },
            },
        },
    })),
}
