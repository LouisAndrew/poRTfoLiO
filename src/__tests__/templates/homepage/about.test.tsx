import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'

import { render, cleanup } from '@testing-library/react'
import '@testing-library/jest-dom'

import About from 'templates/homepage/about'

import * as Gatsby from 'gatsby'

const useStaticQuery = jest.spyOn(Gatsby, 'useStaticQuery')

const mockAboutcontent = 'Hello, World!'

describe('About', () => {
    const Element = <About />

    afterEach(cleanup)
    beforeEach(() => {
        useStaticQuery.mockImplementationOnce(() => ({
            aboutData: {
                frontmatter: {
                    about: {
                        about_content: mockAboutcontent,
                    },
                },
            },
            imgS: {
                frontmatter: {
                    about: {
                        about_imgs: [
                            {
                                childImageSharp: {
                                    fixed: {},
                                },
                            },
                        ],
                    },
                },
            },
            imgM: {
                frontmatter: {
                    about: {
                        about_imgs: [
                            {
                                childImageSharp: {
                                    fixed: {},
                                },
                            },
                        ],
                    },
                },
            },
            imgL: {
                frontmatter: {
                    about: {
                        about_imgs: [
                            {
                                childImageSharp: {
                                    fixed: {},
                                },
                            },
                        ],
                    },
                },
            },
            imgXL: {
                frontmatter: {
                    about: {
                        about_imgs: [
                            {
                                childImageSharp: {
                                    fixed: {},
                                },
                            },
                        ],
                    },
                },
            },
        }))
    })

    it('renders without crashing', () => {
        const div = document.createElement('div')
        ReactDOM.render(Element, div)
    })

    it('Should render the "about me" content correctly', () => {
        const { getByTestId } = render(Element)

        const aboutMeContent = getByTestId('content')

        expect(aboutMeContent).toHaveTextContent(mockAboutcontent)
    })

    it('matches snapshot', () => {
        const run = true

        expect(run).toBeTruthy()

        const tree = renderer.create(Element).toJSON()
        expect(tree).toMatchSnapshot()
    })
})
