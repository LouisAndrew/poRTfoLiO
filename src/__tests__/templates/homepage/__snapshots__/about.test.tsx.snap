// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About matches snapshot 1`] = `
<div
  className="css-zkfaav"
  id="about"
>
  <div
    className="css-l3rx45"
  >
    <h2
      className="css-1hc7vq6"
    >
      About me
    </h2>
    <div
      className="css-ci7kx9"
    >
      <div
        className=" gatsby-image-wrapper"
        style={
          Object {
            "display": "inline-block",
            "flexShrink": 0,
            "height": undefined,
            "overflow": "hidden",
            "position": "relative",
            "width": undefined,
          }
        }
      >
        <picture>
          <source
            media="(max-width: 640px)"
          />
          <source
            media="(max-width: 960px)"
          />
          <source
            media="(min-width: 1952px)"
          />
          <source />
          <img
            alt="portrait of me"
            loading="lazy"
            onLoad={[Function]}
            style={
              Object {
                "borderRadius": "50%",
                "flexShrink": 0,
                "height": "100%",
                "left": 0,
                "objectFit": "cover",
                "objectPosition": "center",
                "opacity": 0,
                "position": "absolute",
                "top": 0,
                "transition": "opacity 500ms",
                "width": "100%",
              }
            }
          />
        </picture>
        <noscript
          dangerouslySetInnerHTML={
            Object {
              "__html": "<picture><source media=\\"(max-width: 640px)\\" srcset=\\"undefined\\" /><source media=\\"(max-width: 960px)\\" srcset=\\"undefined\\" /><source media=\\"(min-width: 1952px)\\" srcset=\\"undefined\\" /><source srcset=\\"undefined\\" /><img loading=\\"lazy\\" src=\\"\\" alt=\\"portrait of me\\" style=\\"position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center\\"/></picture>",
            }
          }
        />
      </div>
      <p
        className="css-lho9bf"
        data-testid="content"
      >
        Hello, World!
      </p>
    </div>
  </div>
</div>
`;
